src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,90 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private T[] array = (T[]) new Object[10];
+    private int size = 0;
```

Comment:
* array should be initialized in the constructor
* primitive field types have default values
* 10 is a magic number